File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length-1;
        int sum = numbers[l]+numbers[r];
        while(sum !=target){
            if(sum>target){
                r = r-1;
            }else{
                l = l+1;
            }
            sum = numbers[l]+numbers[r];
            System.out.println(sum);
        }
        return new int[]{l+1,r+1};

        
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        let subMap = new Map()
        for(let i = 0; i< nums.length; i++){
            if(subMap.has(nums[i])){
                return[subMap.get(nums[i]), i]
            }else{
                console.log(subMap)
                subMap.set(target-nums[i], i)
            }
        }
        return [];
    }
}
